core/router/router.go

```diff
@@ -93,10 +94,8 @@ func (r *router) Init(c *component.Component) error {
 	}()
 	r.Component.SetStatus(component.StatusHealthy)
 	if r.Component.Monitor != nil {
-		r.monitorStream = r.Component.Monitor.NewRouterStreams(r.Identity.Id, r.AccessToken)
-		go r.Component.Monitor.TickStatus(func() {
-			r.monitorStream.Send(r.GetStatus())
-		})
+		r.monitorStream = r.Component.Monitor.RouterClient(r.Context, grpc.PerRPCCredentials(auth.WithStaticToken(r.AccessToken)))
+		go r.Component.Monitor.TickStatus(func() { r.monitorStream.Send(r.GetStatus()) })
```

Comment:
this goroutine never returns. Which may cause trouble. What if the component is down/disconnected? Do we need to query it's status all the time and send it?